client/blocks/login/index.jsx

```diff
@@ -40,12 +40,30 @@ class Login extends Component {
 	};
 
 	componentWillMount = () => {
-		if ( ! this.props.twoStepNonce && this.props.twoFactorAuthType && typeof window !== 'undefined' ) {
+		if ( typeof window === 'undefined' ) {
```

Comment:
> This is the only lifecycle hook called on server rendering.

You can use `componentDidMount` and check against `window` won't be necessary. See https://facebook.github.io/react/docs/react-component.html#componentwillmount